src/jit/lsra.cpp

```diff
@@ -5312,7 +5312,7 @@ LinearScan::spillInterval(Interval* interval, RefPosition* fromRefPosition, RefP
 
     if (!fromRefPosition->lastUse)
     {
-        if (!fromRefPosition->IsActualRef())
+        if (!fromRefPosition->RequiresRegister())
         {
```

Comment:
I think we are still not seeing eye-to-eye on this. I don’t think it is reasonable to have `RequiresRegister()` return true for a RefPosition that may not get a register. It just doesn’t make sense, and could easily lead to confusion.
I think that this condition should remain as-is - because that's the case where, even if it does not require a register, we must treat it as spilled if we had previously allocated one.